cli/src/commands/git/push.rs

```diff
@@ -295,17 +300,56 @@ pub fn cmd_git_push(
         return Ok(());
     }
 
-    validate_commits_ready_to_push(ui, &bookmark_updates, &remote, &tx, args)?;
+    let sign_behavior = if tx.settings().get_bool("git.sign-on-push")? {
+        Some(SignBehavior::Own)
+    } else {
+        None
+    };
+    let commits_to_sign =
+        validate_commits_ready_to_push(ui, &bookmark_updates, &remote, &tx, args, &sign_behavior)?;
+    let num_updated_signatures = commits_to_sign.len();
+
```

Comment:
Maybe we'll need `.check_rewritable()` somewhere (or exclude immutable commits)?

Reply:
Oh, never mind. It appears to be excluded by `.union(workspace_helper.env().immutable_heads_expression())`.

Reply:
Hmm, is it actually? I thought given it's just a union and not an intersection with the old heads, there might be some items which are descendants of the old heads but still immutable.

Reply:
It's `(old_heads | immutable_heads)..new_heads` so I think `commits_to_push` are actually `mutable_commits_to_push`. In any case, adding tests will help.

There was inline comment in the original patch 6ae4f458, but apparently it was lost.